pkg/analysis_server/test/services/correction/fix_test.dart

```diff
@@ -5897,6 +5897,36 @@ class Sub extends Test {
 ''');
   }
 
+  test_lint_avoid_init_to_null_top_level() async {
+    String src = '''
+var /*LINT*/x = null;
+''';
+    await findLint(src, LintNames.avoid_init_to_null);
+
+    await applyFix(DartFixKind.LINT_REMOVE_INIT_TO_NULL);
+
+    verifyResult('''
+var x;
+''');
+  }
+
+  test_lint_avoid_init_to_null_field() async {
+    String src = '''
+class Test {
+  int /*LINT*/x = null;
+}
+''';
+    await findLint(src, LintNames.avoid_init_to_null);
+
+    await applyFix(DartFixKind.LINT_REMOVE_INIT_TO_NULL);
+
+    verifyResult('''
+class Test {
+  int x;
+}
+''');
+  }
+
```

Comment:
We should have at least one test where the initializer is in the middle of a list of variable declarations. Perhaps something like:
```
void f() {
  String a = 'a', b = null, c = 'c';
}
```
(which will also test local variables, covering another missing case).